test/integration/build/test_build.py

```diff
@@ -80,6 +81,272 @@ def test_app_without_rockspec(cartridge_cmd, project_without_dependencies):
+        'Running `tarantoolctl rocks make {}`'.format(rockspec_name),
+        'Application was successfully built',
+    ]
+
+    # with --spec
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        rockspec_name,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+    assert prebuild_output not in output
+
+    # check that all expected rocks was installed
+    files = recursive_listdir(project.path)
+    assert '.rocks' in files
+    assert all([rock in files for rock in rocks_content])
+
+
+def test_building_with_two_rockspec_in_project_root(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    prebuild_output = "pre-build hook output"
+    version = 'scm-2'
+    second_rockspec_name = '{}-{}.rockspec'.format(project.name, version)
+
+    with open(os.path.join(project.path, second_rockspec_name), 'w') as f:
+        f.write('''
+                package = '{}'
+                version = '{}'
+                source  = {{ url = '/dev/null' }}
+                dependencies = {{ 'tarantool' }}
+                build = {{ type = 'none' }}
+            '''.format(project.name, version))
+
+    with open(os.path.join(project.path, 'cartridge.pre-build'), 'w') as f:
+        prebuild_script_lines = [
+            "#!/bin/sh",
+            "echo \"{}\"".format(prebuild_output)
+        ]
+        f.write('\n'.join(prebuild_script_lines))
+
+    build_logs = [
+        'Build application in',
+        'Running `cartridge.pre-build`',
+        'Running `tarantoolctl rocks make`',
+        'Application was successfully built',
+    ]
+
+    # without --spec
+    cmd = [
+        cartridge_cmd,
+        "build",
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+    assert prebuild_output not in output
+
+    # check that all expected rocks was installed
+    files = recursive_listdir(project.path)
+
+    # tarantoolctl performs build with the oldest version of rockspec files
+    rocks_content = get_base_project_rocks(project.name, second_rockspec_name, version)
+
+    assert '.rocks' in files
+    assert all([rock in files for rock in rocks_content])
+
+    build_logs = [
+        'Build application in',
+        'Running `cartridge.pre-build`',
+        'Running `tarantoolctl rocks make {}`'.format(second_rockspec_name),
+        'Application was successfully built',
+    ]
+
+    # with --spec and .rockspec file in project root
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        second_rockspec_name,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+    assert prebuild_output not in output
+
+    # check that all expected rocks was installed
```

Comment:
We can check that specified rockspec was used a little tricky.

We can use [command build type](https://github.com/luarocks/luarocks/wiki/Rockspec-format#command) in rockspec and ask it to say who it is. Then, just check which rockspec was called by output.

```lua
build = {
    type = 'command',
    build_command = 'echo I am scm-1',
}```

Note, that you should call `cartridge build --verbose` to handle build logs.